### PR TITLE
Prevent NaN gradients, setup package organization, add regression module

### PR DESCRIPTION
**Pull request by elfprince13**
This is a bit of a kitchen sink PR:
- Moves `fqs.py` into a proper package and sets up Python project files so this can be used as an external dependency
- Ensures that the root finding operations are all safely differentiable, which is the main point of having a TF implementation anyway.
- Adds and documents a module that aids in solving orthogonal (also called "total") least-quartic linear regressions.

### CODE REVIEW

File: src/fqs/test_fqs.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
+from typing import Any, Callable, FrozenSet, Set
+
```
Comment (tnybny) [inline src/fqs/test_fqs.py:9]: You mentioned you wanted to add authorship info?

File: src/fqs/test_fqs.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
+from typing import Any, Callable, FrozenSet, Set
```
Comment (tnybny) [inline src/fqs/test_fqs.py:8]:
```suggestion
#!/usr/bin/env python3
from typing import Any, Callable, FrozenSet, Set
```

File: src/fqs/test_orthogonal_least_quartics.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+from .fqs import quartic
```
Comment (tnybny) [inline src/fqs/test_orthogonal_least_quartics.py]:
```suggestion
#!/usr/bin/env python3
from .fqs import quartic
```

File: src/fqs/test_orthogonal_least_quartics.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        self.assertIn(True, tf_isclose(roots, 0., atol=1e-6))
+
+    def test_square(self) -> None:
+        x = tf.constant([-1, -1, -1, 0, 0, 0, 1, 1, 1], dtype=tf.float32)
+        y = tf.constant([-1, 0, 1, -1, 0, 1, -1, 0, 1], dtype=tf.float32)
+        roots = _run_one_test(x, y)
+        self.assertIn(True, tf_isclose(roots, 0., atol=1e-6))
+
+if __name__ == "__main__":
+    tf.config.run_functions_eagerly(True)
```
Comment (tnybny) [inline src/fqs/test_orthogonal_least_quartics.py]: did we mean to leave this to True?

File: src/fqs/fqs.py [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    Returns
+    -------
+    roots: tuple
+        Output data is a tuple of three roots of a given polynomial.
+    '''
+    with tf.name_scope("cubic"):
+        if not assume_cubic:
+            (quadratic_mask, a0) = _safe_denom(a0)
+        #if unique_real_root and (a0.dtype.is_complex or b0.dtype.is_complex or c0.dtype.is_complex or d0.dtype.is_complex):
+        #    raise TypeError("There may be no real root if the coefficients are complex")
```
Comment (tnybny) [inline src/fqs/fqs.py]:
```suggestion
```

Comment (elfprince13) [reply src/fqs/fqs.py]: Addressed in [4b94568](https://github.com/Geopipe/fqs/pull/1/commits/4b94568e86e1d7ca1f66a29dffb5a0a867b8bb95)

File: src/fqs/test_orthogonal_least_quartics.py (hunk 4 above)
Comment (elfprince13) [reply src/fqs/test_orthogonal_least_quartics.py]: Oops. Fixed in [4b94568](https://github.com/Geopipe/fqs/pull/1/commits/4b94568e86e1d7ca1f66a29dffb5a0a867b8bb95).

File: src/fqs/test_fqs.py (hunk 1 above)
Comment (elfprince13) [reply src/fqs/test_fqs.py:9]: Resolved in [4b94568](https://github.com/Geopipe/fqs/pull/1/commits/4b94568e86e1d7ca1f66a29dffb5a0a867b8bb95)

File: src/fqs/test_fqs.py (hunk 2 above)
Comment (elfprince13) [reply src/fqs/test_fqs.py:8]: More flexible execution handled in [094e344](https://github.com/Geopipe/fqs/pull/1/commits/094e344674ae5afb774364f1ccfba4b1893453ad).

**Review by tnybny — APPROVED** [review]

**Pull request merged**